local-dev/run_postgres.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+# Start up a postgres container with initial user/database setup.
+POSTGRES_VERSION=9.6
+start() {
+    echo "attempting to remove old $CONTAINER container..."
+    docker rm -f $CONTAINER || echo "docker rm failed. nothing to rm."
+
+    # start up postgres
+    echo "starting up postgres container..."
+    docker run --rm  --name $CONTAINER -e POSTGRES_PASSWORD=password -p "$POSTGRES_PORT:5432" \
+      -v $PWD/local-dev/local-postgres-init.sql:/docker-entrypoint-initdb.d/docker_postgres_init.sql \
```

Comment:
Bind mounting is quite slow on mac and has a running cost, unfortunately; is there some other way we can attach files?

Reply:
Sorry, I'm not developing on mac, so I'm going to need help testing that any changes are good ones. I'd hope there are easy solutions here.
Is it the initial mounting that's slow (even for a single tiny file), or write propagation between host/container, or just the existence of the connection? I don't imagine that we mount many things, or that this file is something that gets modified during development. How slow is it to start up now? What's the constant running cost?

Reply:
The issue is that, as long as the mount is active, everything chugs. It's an issue with macOS. As an alternative, consider:

```bash
docker create --name $CONTAINER -e POSTGRES_PASSWORD=password -p "$POSTGRES_PORT:5432" postgres:$POSTGRES_VERSION

docker cp $PWD/local-dev/local-postgres-init.sql $CONTAINER/docker-entrypoint-initdb.d/docker_postgres_init.sql

docker run --rm -d $CONTAINER
```

Also, is there a reason we have to spin up a second container to run the validation?

Reply:
Thanks, PTAL.
I think the original reasoning is that using a second container to run the validation allows the whole thing to start up without postgres being installed on the host. You could do all of your connections to the DB through docker, and never worry about having postgres installed on your machine. I don't have a strong opinion about this.

Reply:
Side note on Docker performance with mounted volumes on OSX. This is something AoU struggled with quite a bit, since we used Docker to kick off our entire local API server. The performance degradation mostly seemed to scale with the # of files being mounted and how frequently they were being written to. (For example, we were blindly mounting the entire project directory, including the `build` subdirectory where intermediate outputs were constantly getting rewritten. This was bad.)

For a single-file mount like here, I wouldn't expect the performance hit to be too large. But just copying the script file makes sense as a best practice, where possible.

Reply:
@gjuggler one solution we've used for that in the past is to use rsync with an intermediate container, since it doesn't have a performance penalty.

Reply:
ack – the AoU team eventually adopted http://docker-sync.io/ which packages up a similar approach.